src/lib/helpers.js

```diff
@@ -0,0 +1,596 @@
+}
+
+function ipIsPrivateOrLocalhost(ip) {
+    if (ip.substring(0, 7) === "::ffff:")
+        ip = ip.substring(7);
+
+    if (net.isIPv4(ip)) {
+        // check localhost
+        if (ip === '127.0.0.1')
+            return true;
+
+        //         10.0.0.0 - 10.255.255.255        ||   172.16.0.0 - 172.31.255.255                          ||    192.168.0.0 - 192.168.255.255
+        return /^(10)\.(.*)\.(.*)\.(.*)$/.test(ip) || /^(172)\.(1[6-9]|2[0-9]|3[0-1])\.(.*)\.(.*)$/.test(ip) || /^(192)\.(168)\.(.*)\.(.*)$/.test(ip)
+    }
+
+    // else: ip is IPv6
+    const firstWord = ip.split(":").find(el => !!el); //get first not empty word
+
+    // equivalent of 127.0.0.1 in IPv6
+    if (ip === "::1")
+        return true;
+
+    // The original IPv6 Site Local addresses (fec0::/10) are deprecated. Range: fec0 - feff
+    else if (/^fe[c-f][0-f]$/.test(firstWord))
+        return true;
+
+    // These days Unique Local Addresses (ULA) are used in place of Site Local.
+    // Range: fc00 - fcff
+    else if (/^fc[0-f]{2}$/.test(firstWord))
+        return true;
+
+    // Range: fd00 - fcff
+    else if (/^fd[0-f]{2}$/.test(firstWord))
+        return true;
+
+    // Link local addresses (prefixed with fe80) are not routable
+    else if (firstWord === "fe80")
+        return true;
+
+    // Discard Prefix
+    else if (firstWord === "100")
+        return true;
+
+    // Any other IP address is not Unique Local Address (ULA)
+    return false;
+}
+
+function isHexStringOfLength(str, length) {
+    return (isHexString(str)
+        && (
+            (str.starsWith('0x') && str.slice(2).length === length * 2)
+            || str.length === length * 2
+        )
+    );
+}
+
+function isHexString(str) {
+    return (
+        typeof str === 'string'
+        && (
+            (str.startsWith("0x") && /^[a-fA-F0-9]+$/i.test(str.slice(2)))
+            || /^[a-fA-F0-9]+$/i.test(str)
+        )
+    );
+}
+
+function toHexString(buffer) {
+    return "0x" + Array.prototype.map.call(buffer, x => ('00' + x.toString(16)).slice(-2))
+        .join('')
+        .match(/[a-fA-F0-9]{2}/g)
+        .join('')
+}
+
+function parseURL(url) {
+    if (url && typeof url === 'string' && url.indexOf("://") > -1) {
+        const hostIndex = url.indexOf("://") + 3
+        const schema = url.slice(0, hostIndex)
+        let host = url.slice(hostIndex)
+        let path = ""
+        let query = ""
+        const pathIndex = host.indexOf("/")
+        if (pathIndex > -1) {
+            path = host.slice(pathIndex + 1)
+            host = host.slice(0, pathIndex)
+            const queryIndex = path.indexOf("?")
+            if (queryIndex > -1) {
+                query = path.slice(queryIndex + 1)
+                path = path.slice(0, queryIndex)
+            }
+        }
+        return [schema, host, path, query];
+    } else {
+        throw new TypeError(`Invalid URL was provided: ${url}`)
+    }
+}
```

Comment:
This should be equivalent
```suggestion
function parseURL(url) {
    try {
        const parsedUrl = new URL(url)
        return [parsedUrl.protocol + "//", parsedUrl.host, parsedUrl.pathname.slice(1), parsedUrl.search.slice(1)]
    } catch {
        throw new TypeError(`Invalid URL was provided: ${url}`)
    }
}
```